container/pull.bzl

```diff
@@ -84,20 +103,15 @@ container_import(
     if result.return_code:
         fail("Pull command failed: %s (%s)" % (result.stderr, " ".join(args)))
 
-container_pull = repository_rule(
-    attrs = {
-        "registry": attr.string(mandatory = True),
-        "repository": attr.string(mandatory = True),
-        "digest": attr.string(),
-        "tag": attr.string(default = "latest"),
-        "_puller": attr.label(
-            executable = True,
-            default = Label("@puller//file:downloaded"),
-            cfg = "host",
-        ),
-    },
-    implementation = _impl,
-)
+    updated_attrs = {
+        k: getattr(repository_ctx.attr, k)
+        for k in _container_pull_attrs.keys()
+    }
+    digest_result = repository_ctx.execute(["cat", repository_ctx.path("image/digest")])
```

Comment:
lets avoid using execute in repository rules as much as possible (particularly for things like this) as this will not be platform agnostic

Reply:
we already use `execute` to call the puller, but yeah, I'm a bit unhappy about having to shell out to `cat`.

I couldn't find another way to read a file into a starlark variable, though. If we don't like `cat`, I guess we could write a small python wrapper, though is that better or worse than using `cat`?

Reply:
ugh, using a python wrapper properly is too much effort, alright, lets leave as is then. This PR looks good overall then as is. Lets try to get the upstream change. Also, could you try to add a test (we can wait till upstream change is in if you want)